leptonai/photon/worker.py

```diff
@@ -0,0 +1,130 @@
+import asyncio
+from abc import abstractmethod
+import datetime
+from enum import Enum
+import json
+import threading
+import time
+from typing import Any, Dict
+import uuid
+
+from loguru import logger
+
+from .photon import Photon, HTTPException
+from leptonai.queue import Queue, Empty
+from leptonai.kv import KV
+from leptonai.util import asyncfy_with_semaphore
+
+
+class Status(str, Enum):
+    CREATED = "CREATED"
+    RUNNING = "RUNNING"
+    SUCCESS = "SUCCESS"
+    FAILED = "FAILED"
+
+
+class Worker(Photon):
+    queue_name: str = None
+    kv_name: str = None
+
+    queue_empty_sleep_time: int = 5
+    save_result: bool = False
+
+    LEPTON_TASK_ID_TAG = "_lepton_task_id"
+
+    def init(self):
+        super().init()
+
+        if self.queue_name is None:
+            raise RuntimeError("queue_name is not set")
+        self._queue = Queue(
+            self.queue_name, create_if_not_exists=False, wait_for_creation=True
+        )
+        if self.kv_name is None:
+            raise RuntimeError("kv_name is not set")
+        self._kv = KV(self.kv_name, create_if_not_exists=False, wait_for_creation=True)
+
+        # so that `on_task` shares the handler_max_concurrency
+        self._on_task_handler = asyncfy_with_semaphore(
+            self.on_task, self._handler_semaphore
+        )
+
+        threading.Thread(target=self._worker_loop, daemon=True).start()
```

Comment:
Should we not do daemon=True and allow the last job to have a chance to finish? Basically, we can set a should_exit member variable and periodically check it, and if some grace period ends, kill the task. Or better, we should record the current_task_ids and send back to the kv saying "this task timed out"